libraries/chain/abi_serializer.cpp

```diff
@@ -193,7 +193,7 @@ namespace eosio { namespace chain {
       return itr->second;
    }
 
-   void abi_serializer::validate()const {
+   void abi_serializer::validate(const fc::microseconds& max_serialization_time)const {
```

Comment:
Called by `set_abi()` but calculates its own deadline.  Should instead accept a deadline.